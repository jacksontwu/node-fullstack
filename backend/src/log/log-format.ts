const formatError = (ctx: any, err: any) => {
    const method = ctx.method;
    const url = ctx.url;
    const body = ctx.request.body;
    const userAgent = ctx.header.userAgent;
    return { method, url, userAgent, body, err };
};

const formatReq = (ctx: any) => {
    const method = ctx.method;
    const url = ctx.url;
    const body = ctx.query;
    const header = ctx.header;
    return { method, url, header, body };
};

const formatRes = (ctx: any, costTime: number) => {
    const method = ctx.method;
    const url = ctx.url;
    const body = ctx.request.body;
    const header = ctx.header;
    const response = ctx.response;
    return { method, url, header, body, costTime, response };
};

export { formatError, formatReq, formatRes };
