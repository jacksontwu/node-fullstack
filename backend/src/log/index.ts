import config from './log4js';
import { errLogger, resLogger, reqLogger } from './logger';

export { config, errLogger, reqLogger, resLogger };
