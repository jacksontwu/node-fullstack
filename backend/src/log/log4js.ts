import { Configuration } from 'log4js';
import { join } from 'path';
import { logDir } from 'src/constants';

const config: Configuration = {
    appenders: {
        error: {
            type: 'file',
            category: 'errLogger',
            filename: join(logDir, 'error.log'),
            maxLogSize: 104800,
            backups: 100,
        },
        default: {
            type: 'file',
            category: 'defaultLogger',
            filename: join(logDir, 'default'),
            pattern: 'yyyy-MM-dd.log',
            alwaysIncludePattern: true,
            maxLogSize: 104800,
            backups: 100,
        },
    },
    categories: {
        error: { appenders: ['error'], level: 'error' },
        default: { appenders: ['default'], level: 'info' },
    },
};

export default config;
