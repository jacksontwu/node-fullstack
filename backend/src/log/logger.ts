import { formatError, formatRes, formatReq } from './log-format';
import { getLogger } from 'log4js';

const errorLogger = getLogger('error');
const logger = getLogger('default');

const errLogger = (ctx: any, error: any) => {
    if (ctx && error) {
        errorLogger.error(formatError(ctx, error));
    }
};
const resLogger = (ctx: any, resTime: number) => {
    if (ctx) {
        logger.info(formatRes(ctx, resTime));
    }
};

const reqLogger = (ctx: any) => {
    if (ctx) {
        logger.info(formatReq(ctx));
    }
};

export { errLogger, resLogger, reqLogger };
