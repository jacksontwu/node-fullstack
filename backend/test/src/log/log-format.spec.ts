import { formatError, formatReq, formatRes } from '../../../src/log/log-format';

describe('logger', () => {
    it('formatError', () => {
        const ctx = {
            method: 'a',
            url: 'b',
            query: 'c',
            header: 'd',
        };
        expect(formatError(ctx, 1)).toMatchSnapshot();
    });
    it('formatReq', () => {
        const ctx = {
            method: 'a',
            url: 'b',
            query: 'c',
            header: 'd',
        };
        expect(formatReq(ctx)).toMatchSnapshot();
    });
    it('formatRes', () => {
        const ctx = {
            method: 'a',
            url: 'b',
            query: 'c',
            header: 'd',
            response: 'e',
        };
        expect(formatRes(ctx, 1)).toMatchSnapshot();
    });
});
