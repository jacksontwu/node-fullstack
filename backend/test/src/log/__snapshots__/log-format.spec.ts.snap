// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logger formatError 1`] = `
Object {
  "body": "c",
  "err": 1,
  "header": "d",
  "method": "a",
  "url": "b",
}
`;

exports[`logger formatReq 1`] = `
Object {
  "body": "c",
  "header": "d",
  "method": "a",
  "url": "b",
}
`;

exports[`logger formatRes 1`] = `
Object {
  "body": "c",
  "costTime": 1,
  "header": "d",
  "method": "a",
  "response": "e",
  "url": "b",
}
`;
